### ISSUE TITLE
[PRO]: extension-unique-id does not maintain ids

### ISSUE THREAD

**Issue by arnaud-secondlayer** [ueberdosis/tiptap#3974]
### Which packages did you experience the bug in?

extension-unique-id

### What Tiptap version are you using?

2.0.0

### What’s the bug you are facing?

I use `setContent` with the following content (already containing unique ids in field `uid`):

```
{
    "type": "doc",
    "content": [
        {
            "type": "title",
            "attrs": {
                "uid": "d23ac884-4609-49a4-a5a4-53d840fdd3cf",
                "name": "title",
                "class": "title"
            },
            "content": [
                {
                    "type": "text",
                    "text": "Title"
                }
            ]
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "9f9d65aa-8b17-49be-a4e0-d2a386351d6b"
            },
            "content": [
                {
                    "type": "text",
                    "text": "this is working"
                }
            ]
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "bbe397c0-8088-4cb2-9851-ee32bebd3312"
            }
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "c28cf653-531c-42d2-ad15-3efdb4a97c53"
            }
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "9f477b6e-d084-4916-899c-3ab452f6622e"
            }
        }
    ]
}
```

When retrieving the content with `editor.getJSON()`, I can notice that the final ids of the empty paragraph have changed:

```
{
    "type": "doc",
    "content": [
        {
            "type": "title",
            "attrs": {
                "uid": "d23ac884-4609-49a4-a5a4-53d840fdd3cf",
                "name": "title",
                "class": "title"
            },
            "content": [
                {
                    "type": "text",
                    "text": "Title"
                }
            ]
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "9f9d65aa-8b17-49be-a4e0-d2a386351d6b"
            },
            "content": [
                {
                    "type": "text",
                    "text": "this is working"
                }
            ]
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "1c5f884d-8421-419f-a9a6-dfd3a989ba14"
            }
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "6cf3e4d5-590b-4a4c-b0f4-8a0c00ca4a34"
            }
        },
        {
            "type": "paragraph",
            "attrs": {
                "uid": "bbe397c0-8088-4cb2-9851-ee32bebd3312"
            }
        }
    ]
}
```

### What browser are you using?

Chrome

### What did you expect to happen?

I would have expected the ids to remain unchanged because setting the content with all nodes already have a unique id.
I'm also not sure to understand why id `bbe397c0-8088-4cb2-9851-ee32bebd3312` is changing position in the document.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by Jackesa** [comment]
This is causing trouble for my team as well.

**Comment by joaopcm** [comment]
Same here...

**Comment by shuyudao** [comment]
Same

If the paragraph is preceded by an empty paragraph, the empty paragraph generates a new uid even if the uid attribute has been set, and the first paragraph with content after the empty paragraph generates a new uid if there is a content change.


"@tiptap-pro/extension-unique-id": "^2.1.0",
"@tiptap/pm": "^2.0.3",
"@tiptap/react": "^2.0.3",
"@tiptap/starter-kit": "^2.0.3",

**Comment by lin52025iq** [comment]
A similar problem occurs when a line break is copied and pasted to the top of a paragraph. The uuid of the paragraph before pasting is updated.

https://github.com/ueberdosis/tiptap/assets/87741822/3bbbcce4-7e73-49ef-8133-8319337114e3

**Comment by olavhaugen** [comment]
I have this issue as well. Any update on this?

**Comment by andyszy** [comment]
FWIW, my team was experiencing this issue a couple months ago, but we can no longer repro it. No promises, but this may have been fixed :)

**Comment by bdbch** [comment]
Thanks, will take a look into it.

**Comment by bdbch** [comment]
Found the issue and opening a PR now

**Comment by iarmankhan** [comment]
@bdbch Any updates here? We are also facing the similar issue with uniqueId being changed when using yjs

**Comment by nperez0111** [comment]
This has been released with `v2.10.11`

**Comment by dan-cooke** [comment]
@nperez0111  this is still an issue for tables in `2.11.1` shall we re-open or raise a new bug?

**Comment by Pigzhuyao** [comment]
@dan-cooke May I ask if the primary key of your new paragraph is different from that of the previous paragraph?

**Comment by Haraldson** [comment]
@olavhaugen Are you still using this workaround or did recent developments here fix the issue for you? We’re still having issues, especially in the “empty paragraph” scenario mentioned above.

**Comment by olavhaugen** [comment]
@Haraldson Yes, I am still using that workaround, but I haven't touched that code in a while.

That said, I noticed that after upgrading from Tiptap 2 -> 3, I had some issues with IDs being generated when I used the [setContent](https://tiptap.dev/docs/editor/api/commands/content/set-content#options) command. In our case, that was because Tiptap 3 changed the default value of the `emitUpdate` property from `false` to `true`. Explicitly setting it to `setContent({...}, { emitUpdate: false })` solved a similar issue for us. 

Might be worth looking into 🤷‍♂️
